src/main/scala/circt/stage/Annotations.scala

```diff
@@ -137,3 +137,17 @@ private[circt] case object SplitVerilog extends NoTargetAnnotation with CIRCTOpt
   )
 
 }
+
+/** Write the intermediate `.fir` file in [[circt.stage.ChiselStage]]
+  */
+case object DumpFir extends NoTargetAnnotation with CIRCTOption with HasShellOptions {
```

Comment:
For truly new annotations, consider making these package private and force users to go through the CLI-level APIs.

Reply:
I've kind of waffled on this, for users who have their own command line parsers, they need the `HasShellOptions` objects to be public to build their own CLI. That being said, I'm not sure what use case there would be to use this option on its own rather than just using all of `circt.stage.CLI` so I think making this one private is reasonable. Options to Chisel generation probably need to remain public but that is not what this is!